lib/tbot/ca_rotation.go

```diff
@@ -114,6 +117,18 @@ func (b *Bot) caRotationLoop(ctx context.Context) error {
 			return nil
 		}
 
+		// If the error is due to the client being replaced with a new client
+		// as part of the credentials renewal. Ignore it, and immediately begin
+		// watching again with the new client. We can safely check for Canceled
+		// here, because if the context was actually canceled, it would've
+		// been caught in the error check immediately following watchCARotations
+		var statusErr interface {
+			GRPCStatus() *status.Status
+		}
+		if errors.As(err, &statusErr) && statusErr.GRPCStatus().Code() == codes.Canceled {
+			continue
+		}
```

Comment:
```suggestion
		if statusErr, _ := status.FromError(err); statusErr.Code() == codes.Canceled {
			continue
		}
```

Reply:
Also, are you sure we shouldn't keep the jitter at least?

Reply:
Agree on the jitter, but wouldn't your suggested change not work with wrapped errors ?

Reply:
Addressed by https://github.com/gravitational/teleport/pull/19266/commits/77ff82c5ba76a74205d30b8c9db9f491b03d9774